apps/web/pages/auth/setup/index.tsx

```diff
@@ -1,47 +1,100 @@
 import { UserPermissionRole } from "@prisma/client";
 import { GetServerSidePropsContext } from "next";
-import { useState } from "react";
+import { Dispatch, SetStateAction, useState } from "react";
 
 import AdminAppsList from "@calcom/features/apps/AdminAppsList";
 import { getSession } from "@calcom/lib/auth";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import prisma from "@calcom/prisma";
 import { inferSSRProps } from "@calcom/types/inferSSRProps";
-import { WizardForm } from "@calcom/ui";
+import { Meta, WizardForm } from "@calcom/ui";
 
-import SetupFormStep1 from "./steps/SetupFormStep1";
-import StepDone from "./steps/StepDone";
+import AdminUser from "@components/setup/AdminUser";
+import ChooseLicense from "@components/setup/ChooseLicense";
+import EnterpriseLicense from "@components/setup/EnterpriseLicense";
 
 export default function Setup(props: inferSSRProps<typeof getServerSideProps>) {
   const { t } = useLocale();
-  const [isLoadingStep1, setIsLoadingStep1] = useState(false);
-  const shouldDisable = props.userCount !== 0;
+  const [isFreeLicense, setIsFreeLicense] = useState(props.deployment?.licenseKey === "");
+  const [isEnabledEE, setIsEnabledEE] = useState(!!props.deployment?.licenseKey);
```

Comment:
We should not read the licenseKey in the frontend. Also if the license KEY is set we cannot choose the FREE version without any explanation. We should allow to select FREE or remove it altogether. It creates unexpected behavior.

Reply:
Fixed, thx!